src/javascript/app/pages/trade/digit_ticker.js

```diff
@@ -0,0 +1,242 @@
+        bottomright_el.classList.add('bottomright');
+        temp_peek_box_el.appendChild(topleft_el);
+        temp_peek_box_el.appendChild(top_el);
+        temp_peek_box_el.appendChild(topright_el);
+        temp_peek_box_el.appendChild(left_el);
+        temp_peek_box_el.appendChild(right_el);
+        temp_peek_box_el.appendChild(bottomright_el);
+        temp_peek_box_el.appendChild(bottomleft_el);
+        temp_peek_box_el.appendChild(bottom_el);
+        temp_peek_box_el.appendChild(temp_mask_el);
+        temp_peek_box_el.appendChild(temp_peek_el);
+
+        const fragment = document.createDocumentFragment();
+        fragment.appendChild(temp_epoch_el);
+        fragment.appendChild(temp_peek_box_el);
+        fragment.appendChild(temp_digits_el);
+        container_element.appendChild(fragment);
+        container_element.classList.add('invisible');
+    };
+
+    // adjust box sizes for mobile
+    const adjustBoxSizes = () => {
+        if (el_container.offsetWidth < 360) {
+            digit_block_size = 28;
+            style_offset_correction = 6;
+        }
+    };
+
+    // Detect winning numbers against the barrier with the given contract type.
+    const getWinningNumbers = (contract_type, barrier) => {
+        switch (contract_type) {
+            case 'DIGITOVER':
+                return array_of_digits.filter(digit => +digit > +barrier);
+            case 'DIGITUNDER':
+                return array_of_digits.filter(digit => +digit < +barrier);
+            case 'DIGITMATCH':
+                return array_of_digits.filter(digit => +digit === +barrier);
+            case 'DIGITDIFF':
+                return array_of_digits.filter(digit => +digit !== +barrier);
+            case 'DIGITODD':
+                return array_of_digits.filter(digit => +digit % 2 !== 0);
+            case 'DIGITEVEN':
+                return array_of_digits.filter(digit => +digit % 2 === 0);
+            default:
+                throw new Error('Cannot Determine Winning numbers.');
+        }
+    };
+
+    const highlightWinningNumbers = (winning_numbers) => {
+        winning_numbers.forEach(digit => {
+            const element = el_container.querySelector(`.digit-${digit}`);
+            element.classList.remove('digit-losing');
+            element.classList.add('digit-winning');
+        });
+    };
+
+    const observeResize = () => {
+        window.onresize = () => {
+            if (el_peek_box) {
+                adjustBoxSizes();
+                el_peek_box.setAttribute('style', `transform: translateX(${calculateOffset()}px)`);
+            }
+        };
+    };
+
+    // Calculate left margin of the peek-box against the "width" of the container
+    const calculateLeftMargin = (width) => (width - (digit_block_size * 10)) / 2;
+
+    // Calculate peek-box left offset.
+    const calculateOffset = () => {
+        const left_offset = calculateLeftMargin(el_container.offsetWidth);
```

Comment:
instead of calculating left margin, how about getting `offsetLeft` of the element `.digit-won`? 
this way, we know the exact position of `.digit-won` in the container with respect to the screen sizes :)

Reply:
Good point.